src/main/java/io/github/oliviercailloux/y2018/apartments/apartment/json/JsonConvert.java

```diff
@@ -71,26 +77,70 @@ private static Path startApartments() {
    */
   @SuppressWarnings("serial")
   public static List<Apartment> jsonToApartments(Path jsonPath) throws IOException {
+    JsonbConfig config = new JsonbConfig().withAdapters(JsonConvert.getAdapter());
+    Jsonb jsonb = JsonbBuilder.create(config);
     String jsonString = Files.readString(jsonPath);
-    List<Apartment.Builder> apartmentsBuild;
     List<Apartment> apartments = new ArrayList<>();
     LOGGER.info("Create ArrayList of Apartment");
-
-    try (Jsonb jsonb = JsonbBuilder.create()) {
+    try (jsonb) {
       LOGGER.info("Create Json builder");
-      apartmentsBuild =
+      apartments =
           jsonb.fromJson(
-              jsonString, new ArrayList<Apartment.Builder>() {}.getClass().getGenericSuperclass());
+              jsonString, new ArrayList<Apartment>() {}.getClass().getGenericSuperclass());
     } catch (Exception e) {
       throw new IllegalArgumentException(e);
     }
-
-    for (Builder apartmentToBuild : apartmentsBuild) {
-      apartments.add(apartmentToBuild.build());
-    }
     return apartments;
   }
 
+  /**
+   * A method that implement a {@link JsonbAdapter} to avoid the use of ApartmentBuilder Code
+   * inspired from the example in the course of Jean-Michel Doudoux available <a>
+   * href="https://www.jmdoudoux.fr/java/dej/chap-json-b.htm">here</a>
+   */
+  public static JsonbAdapter<Apartment, JsonObject> getAdapter() {
+    return new JsonbAdapter<>() {
+      @Override
+      public JsonObject adaptToJson(Apartment apart) throws Exception {
```

Comment:
I'm not sure that `adaptToJson(Apartment apart)` must throws an Exception. (same at line 123)

Reply:
Fix in f2b1f146000c7e0028b3518608dd590dfd693554